File: 566-reshape-the-matrix/566-reshape-the-matrix.cpp
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        
        int row = mat.size();
        int col = mat[0].size();
        
        vector<int> temp;
        vector<vector<int>> reshapedMatrix;
        
        int i = 0;
        int j = 0;
        
        
        if(row*col != r*c) 
            return mat;
        
        
        for(int irow=0; irow<row; irow++)
        {
            for(int icol=0; icol<col; icol++)
            {
                temp.push_back(mat[irow][icol]);
                j += 1;
                if(j == c)
                {
                    reshapedMatrix.push_back(temp);
                    temp.clear();
                    i += 1;
                    j = 0;
                }
            }
        }
        
        return reshapedMatrix;
    }
};

File: 566-reshape-the-matrix/566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        rows: int = len(mat)
        cols: int = len(mat[0])
        
        if rows*cols != r*c:
            return mat
        
        ret_list: List[int] = []
        _: List[int] = []
        i = 0
        for lst in mat:
            for num in lst:
                _.append(num)
                i += 1
                if i == c:
                    i = 0
                    ret_list.append(_)
                    _ = []
        return ret_list
                
                    
        
        
            